Add Android example using Kotlin build scripts

stkent opened the pull request:
Mirrors the existing example Android app, but uses Kotlin build scripts so we can double-check README syntax.

To do this I needed to use a newer-than-stable version of the Android Gradle plugin (3.5.0-alpha09) and Gradle 5.X (5.3.1).

IntelliJ 2019.1 is unable to resolve the build script properly:

<img width="1440" alt="Screen Shot 2019-04-02 at 3 26 21 PM" src="https://user-images.githubusercontent.com/6463980/55430653-ac71c480-555c-11e9-98e9-953317767a4d.png">

but Android Studio 3.5.0-alpha09 is able to (open the example-android-kts project directly):

<img width="1440" alt="Screen Shot 2019-04-02 at 3 27 43 PM" src="https://user-images.githubusercontent.com/6463980/55430664-b267a580-555c-11e9-8152-a7c6cbee4501.png">

stkent committed:
Add Android example using Kotlin build scripts

btkelly committed:
Merge branch 'master' into add-android-kts-example

btkelly commented:
With the updated `.gitignore` I'm seeing a new file included `gnag-plugin.iml` do we really want that included? This is actually a question, I can't remember if the `.iml` files are high churn between developers or if they are more project specific and should be included in source control. Otherwise the new Kotlin example looks good

btkelly committed:
Merge branch 'master' into add-android-kts-example

stkent commented:
I believe it can go either way. E.g from https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems:

> [Directory based project format] is used by all the recent IDE versions by default. Here is what you need to share:
> 
> - [...]
> - **All the .iml module files that can be located in different module directories (applies to IntelliJ IDEA)**

But also:

> You may consider not to share the following:
> 
> - **.iml files and .idea/modules.xml file for the Gradle or Maven based projects, since these files will be generated on import**

Since we're using Gradle, I'll go back to ignoring them. Commit incoming.

stkent committed:
Ignore root iml file

https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
https://github.com/joeblau/gitignore.io/issues/186

stkent commented:
@btkelly updated!
